specification/basic_concepts/basic_concepts.tex

```diff
@@ -38,33 +44,106 @@ \section{Main principles}
-A data type can be used with an arbitrary number of different unregulated port identifiers,
-but not more than one regulated port identifier.
+If a port identifier is specified in a data type definition, such identifier is said to be a
+\emph{fixed port identifier}
+(by inheritance, the same naming principle applies to \emph{fixed subject ID} and \emph{fixed service ID}).
+A data type can be used with an arbitrary number of different port identifiers assigned on a per-application basis,
+but not more than one fixed port identifier.
+
+\subsubsection{Regulation}
+
+Data type definitions can be created by the UAVCAN specification maintainers or by third parties,
+such as vendors or application designers.
+Irrespective of the origin, data types can be included into the set of data type definitions maintained
+and distributed by the UAVCAN specification maintainers;
+definitions belonging to this set are termed \emph{regulated data type definitions}.
+The specification maintainers undertake to keep regulated definitions well-maintained and may occasionally
+amend them and release new versions, if such actions are believed to benefit the ecosystem.
+Data type definitions created by third parties that are not included into the aforementioned set are called
+\emph{unregulated data type definitions}.
+
+Unregulated definitions that are made available for reuse by others are called
+\emph{unregulated public data type definitions};
+those that are kept closed-source for private use by third parties that created them are called
+\emph{(unregulated) private data type definitions}\footnote{The word \emph{``unregulated''} is redundant
+because private data types cannot be regulated, by definition.}.
+
+Data type definitions authored by the specification maintainers for the purpose of supporting and advancing
+this specification are called \emph{standard data type definitions}.
+All standard data type definitions are regulated.
+
+Fixed port identifiers can be used only with regulated data type definitions or with private definitions.
+Fixed port identifiers must not be used with public unregulated data types,
+since that is likely to cause unresolvable port identifier collisions.
+This restriction must be followed at all times by all parties that use the specification;
+if a private data type definition is ever disclosed to any other party (i.e., a party that did not author it)
+or to the public at large, such data type \emph{must be devoid of a fixed port identifier}.
+Equipment and systems that fail to follow this requirement will not be considered UAVCAN-compliant.
+
+\begin{UAVCANSimpleTable}{Data type taxonomy}{|l|X|X|}
+    & Regulated & Unregulated \\
+    \bfseries{Public}
+    &
+    Standard and third-party definitions.\newline
+    Fixed port ID are allowed; they are called \emph{``regulated port ID''}.
+    &
+    Third-party definitions distributed separately from the UAVCAN specification.\newline
```

Comment:
The use of the term "Third-party" makes no sense within this specification given that it is not a contract (i.e. there are no parties). I think we need to say "Definitions not distributed with the UAVCAN specification" here and use a similar strategy for the rest of this section.